lib/spack/docs/contribution_guide.rst

```diff
@@ -0,0 +1,188 @@
+.. _contribution-guide:
+
+Contribution Guide
+=====================
+
+This guide is intended for developers or administrators who want to
+contribute to Spack a bugfix for the package or a new feature.
+It assumes that you have at least some familiarity with Git VCS and Github.
+The guide will show a few examples of contributing workflow and discuss
+the granularity of pull-requests (PRs).
+
+First, what is a PR? Quoting `Bitbucket's tutorials<https://www.atlassian.com/git/tutorials/making-a-pull-request/>`_:
+
+  Pull requests are a mechanism for a developer to notify team members that they have **completed a feature**.
+  The pull request is more than just a notification—it’s a dedicated forum for discussing the proposed feature
+
+Important is completed feature, i.e. the changes one propose in a PR should
+correspond to one feature/bugfix/extension/etc. One can create PRs with
+changes relevant to different ideas, however reviewing such PRs becomes tedious
+and error prone. If possible, try to follow the rule **one-PR-one-package/feature.**
+
+
+Let's assume that the current (patched) state of your fork of Spack is only
+relevant to yourself. Now you come across a bug in a package or would like to
+extend a package and contribute this fix to Spack. It is important that
+whenever you change something that might be of importance upstream,
+create a pull-request (PR) as soon as possible. Do not wait for weeks/months to
+do this: a) you might forget why did you modified certain files; b) it could get
+difficult to isolate this change into a stand-alone clean PR.
+
+Now let us discuss several approaches one may use to submit a PR while
+also keeping your local version of Spack patched.
+
+
+First approach (cherry-picking):
+--------------------------------
+
+First approach is as follows.
+You checkout your local develop branch which for the purpose of this guide
+will be called ``develop_modified``:
+
+.. code-block:: sh
+
+  $git checkout develop_modified
```

Comment:
See #1634. That command isn't a shell script, it's a console session. Use console instead of sh. Also, please put a space after your $ to match the other code-blocks.